From db21ced47fa29887cefe562403d6225bda9441f4 Mon Sep 17 00:00:00 2001
From: MSe1969 <mse1969@posteo.de>
Date: Thu, 30 Aug 2018 17:53:43 +0200
Subject: [PATCH 12/13] Development Settings: Add option to switch off captive
 portal

Add an option to switch off the captive portal, equivalent to adb command
'settings put global captive_portal_mode [1/0]'

Change-Id: Ia3e5b00c370d651b603268d196a67264bc1cc2bf

# Conflicts:
#	res/values/cm_strings.xml
---
 res/values/cm_strings.xml                     |  4 +++
 res/xml/development_prefs.xml                 |  5 ++++
 .../development/DevelopmentSettings.java      | 26 +++++++++++++++++++
 3 files changed, 35 insertions(+)

diff --git a/res/values/cm_strings.xml b/res/values/cm_strings.xml
index cdd13b9818..dbb46301bc 100644
--- a/res/values/cm_strings.xml
+++ b/res/values/cm_strings.xml
@@ -383,4 +383,8 @@
 
     <!-- Screenshot shutter sound -->
     <string name="screenshot_shutter_sound_title">Screenshot shutter sound</string>
+
+    <!-- Captive Portal -->
+    <string name="captive_portal_switch_title">Captive portal mode</string>
+    <string name="captive_portal_switch_summary">Enable or disable the captive portal probing for connection attempts (default ON).</string>
 </resources>
diff --git a/res/xml/development_prefs.xml b/res/xml/development_prefs.xml
index b58221bb6a..dd62451453 100644
--- a/res/xml/development_prefs.xml
+++ b/res/xml/development_prefs.xml
@@ -500,6 +500,11 @@
                 android:title="@string/inactive_apps_title"
                 android:fragment="com.android.settings.fuelgauge.InactiveApps"/>
 
+        <SwitchPreference
+            android:key="captive_portal_switch"
+            android:title="@string/captive_portal_switch_title"
+            android:summary="@string/captive_portal_switch_summary"/>
+
         <SwitchPreference
             android:key="force_allow_on_external"
             android:title="@string/force_allow_on_external"
diff --git a/src/com/android/settings/development/DevelopmentSettings.java b/src/com/android/settings/development/DevelopmentSettings.java
index 3d9164bc37..c7c9630253 100644
--- a/src/com/android/settings/development/DevelopmentSettings.java
+++ b/src/com/android/settings/development/DevelopmentSettings.java
@@ -245,6 +245,8 @@ public class DevelopmentSettings extends RestrictedSettingsFragment
             = "immediately_destroy_activities";
     private static final String APP_PROCESS_LIMIT_KEY = "app_process_limit";
 
+    private static final String CAPTIVE_PORTAL_SWITCH_KEY = "captive_portal_switch";
+
     private static final String BACKGROUND_CHECK_KEY = "background_check";
 
     private static final String SHOW_ALL_ANRS_KEY = "show_all_anrs";
@@ -351,6 +353,8 @@ public class DevelopmentSettings extends RestrictedSettingsFragment
     private SwitchPreference mUSBAudio;
     private SwitchPreference mImmediatelyDestroyActivities;
 
+    private SwitchPreference mCaptivePortalMode;
+
     private ListPreference mAppProcessLimit;
 
     private SwitchPreference mShowAllANRs;
@@ -567,6 +571,9 @@ public class DevelopmentSettings extends RestrictedSettingsFragment
         mAllPrefs.add(mImmediatelyDestroyActivities);
         mResetSwitchPrefs.add(mImmediatelyDestroyActivities);
 
+        mCaptivePortalMode = (SwitchPreference) findPreference(CAPTIVE_PORTAL_SWITCH_KEY);
+        mAllPrefs.add(mCaptivePortalMode);
+
         mAppProcessLimit = addListPreference(APP_PROCESS_LIMIT_KEY);
 
         mShowAllANRs = (SwitchPreference) findPreference(
@@ -891,6 +898,7 @@ public class DevelopmentSettings extends RestrictedSettingsFragment
         updateAnimationScaleOptions();
         updateOverlayDisplayDevicesOptions();
         updateImmediatelyDestroyActivitiesOptions();
+        updateCaptivePortalModeOption();
         updateAppProcessLimitOptions();
         updateShowAllANRsOptions();
         updateShowNotificationChannelWarningsOptions();
@@ -971,6 +979,7 @@ public class DevelopmentSettings extends RestrictedSettingsFragment
         resetDebuggerOptions();
         resetAdbNotifyOptions();
         resetRootAccessOptions();
+        resetCaptivePortalMode();
         writeLogpersistOption(null, true);
         writeLogdSizeOption(null);
         writeAnimationScaleOption(0, mWindowAnimationScale, null);
@@ -1038,6 +1047,12 @@ public class DevelopmentSettings extends RestrictedSettingsFragment
         updateRootAccessOptions();
     }
 
+    private void resetCaptivePortalMode() {
+        Settings.Global.putInt(getActivity().getContentResolver(),
+                        Settings.Global.CAPTIVE_PORTAL_MODE, 1);
+        updateCaptivePortalModeOption();
+    }
+
     private void updateHdcpValues() {
         ListPreference hdcpChecking = (ListPreference) findPreference(HDCP_CHECKING_KEY);
         if (hdcpChecking != null) {
@@ -2396,6 +2411,12 @@ public class DevelopmentSettings extends RestrictedSettingsFragment
                 != 0);
     }
 
+    private void updateCaptivePortalModeOption() {
+        updateSwitchPreference(mCaptivePortalMode, Settings.Global.getInt(
+                getActivity().getContentResolver(), Settings.Global.CAPTIVE_PORTAL_MODE,
+                Settings.Global.CAPTIVE_PORTAL_MODE_PROMPT) != 0);
+    }
+
     private void updateAnimationScaleValue(int which, ListPreference pref) {
         try {
             float scale = mWindowManager.getAnimationScale(which);
@@ -2716,6 +2737,11 @@ public class DevelopmentSettings extends RestrictedSettingsFragment
             writeDisableOverlaysOption();
         } else if (preference == mImmediatelyDestroyActivities) {
             writeImmediatelyDestroyActivitiesOptions();
+        } else if (preference == mCaptivePortalMode) {
+            Settings.Global.putInt(getActivity().getContentResolver(),
+                        Settings.Global.CAPTIVE_PORTAL_MODE,
+                        mCaptivePortalMode.isChecked() ? 1 : 0);
+            updateCaptivePortalModeOption();
         } else if (preference == mShowAllANRs) {
             writeShowAllANRsOptions();
         } else if (preference == mShowNotificationChannelWarnings) {
-- 
2.17.1

